doc/akri_connector/connector-metadata-schema.json

```diff
@@ -0,0 +1,305 @@
+          "type": "string",
+          "description": "The container registry that holds the image for the connector. For example: contoso-acr.io"
+        },
+        "imageName": {
+          "type": "string",
+          "description": "The docker image name for the connector. For example: MyConnector"
+        },
+        "tag": {
+          "type": "string",
+          "description": "The tag of the docker image for this connector"
+        },
+        "digest": {
+          "type": "string",
+          "description": "The digest of the docker image for this connector"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "containerRegistryName",
+            "imageName",
+            "tag"
+          ]
+        },
+        {
+          "required": [
+            "containerRegistryName",
+            "imageName",
+            "digest"
+          ]
+        }
+      ]
+    },
+    "helmConfigurationSettings": {
+      "type": "object",
+      "properties": {
+        "containerRegistryName": {
+          "type": "string",
+          "description": "The container registry name that holds the helm chart for the connector. For example: contoso-acr.io"
+        },
+        "helmChartName": {
+          "type": "string",
+          "description": "The helm chart name for the connector. For example: helm/MyConnector"
+        },
+        "tag": {
+          "type": "string",
+          "description": "The tag of the helm chart for this connector"
+        },
+        "digest": {
+          "type": "string",
+          "description": "The digest of the helm chart for this connector"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "containerRegistryName",
+            "helmChartName",
+            "tag"
+          ]
+        },
+        {
+          "required": [
+            "containerRegistryName",
+            "helmChartName",
+            "digest"
+          ]
+        }
+      ]
+    },
+    "kubernetes": {
+      "type": "object",
+      "description": "The kubernetes-level metadata",
+      "properties": {
+        "minimumKubernetesVersion": {
+          "type": "string",
+          "description": "Minimum required version of kubernetes needed to run this connector"
+        }
+      }
+    },
+    "secrets": {
+      "type": "object",
+      "description": " Secrets that will be mounted onto all connector instances.",
+      "properties": {
+        "secretAlias": {
+          "type": "string",
+          "description": "The application-defined alias for the secret"
+        },
+        "secretKey": {
+          "type": "string",
+          "description": "The key in the secret to be mounted"
+        },
+        "secretRef": {
+          "type": "string",
+          "description": "The name of the secret to be mounted"
+        }
+      },
+      "required": [
+        "secretAlias",
+        "secretKey",
+        "secretRef"
+      ]
+    },
+    "architecture": {
```

Comment:
similar as with secrets we also need trustsettings at Connector and Asset level!